lib/datastore/dataset.js

```diff
@@ -136,7 +135,7 @@ nodeutil.inherits(Dataset, DatastoreRequest);
  * @param {string=} options.apiEndpoint - Custom API endpoint.
```

Comment:
Dataset is no longer a concept for Cloud Datastore. Perhaps we should migrate this to just be a Client object that is used for interacting with a specific project?

(Side note, I tried to attach this comment to L102, but it seems that isn't possible in github?)